scylla-cdc/src/customer.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::HashMap;
+use scylla::frame::response::result::{ColumnSpec, CqlValue, Row};
+
+pub trait Consumer {
+    fn consume_cdc(&self, data: CDCRow);
+}
+
+pub enum OperationType {
+    PreImage = 0,
+    RowUpdate = 1,
+    RowInsert = 2,
+    RowDelete = 3,
+    PartitionDelete = 4,
+    RowRangeDelInclLeft = 5,
+    RowRangeDelExclLeft = 6,
+    RowRangeDelInclRight = 7,
+    RowRangeDelExclRight = 8,
+    PostImage = 9,
+}
+
+pub struct CDCRow {
+    pub stream_id: Vec<u8>,
+    pub time: uuid::Uuid,
+    pub batch_seq_no: i32,
+    pub end_of_batch: bool,
+    pub operation: OperationType,
+    // Can be NULL in the database.
+    pub ttl: Option<i64>,
+    data: Vec<CqlValue>,
+    // Maps element name to its index in the data vector.
+    mapping: HashMap<String, usize>,
+}
+
+const STREAM_ID_NAME: &'static str = "cdc$stream_id";
```

Comment:
I think constants have `static` lifetime by default.

Reply:
That's right, I can't decide which option looks better, so I will just leave it for now.

Reply:
Update: Clippy recommended me to remove it.